src/backend/marsha/core/serializers/playlist.py

```diff
@@ -26,15 +30,41 @@ class Meta:
             "users",
         ]
 
-    def create(self, validated_data):
-        """Create the Playlist object with the passed data."""
-        return Playlist.objects.create(**validated_data)
+    portable_to = serializers.SerializerMethodField(read_only=False)
+
+    def get_portable_to(self, obj):
+        """Getter for portable_to attribute instead of PlaylistSerializer to prevent recursion."""
+        return obj.portable_to.values("id", "title")
+
+    def to_internal_value(self, data):
+        """
+        Validate portable_to.
+
+        Raise a ValidationError if portable_to contains current object playlist id
+        or an unknown playlist id.
+        """
+        value: OrderedDict = super().to_internal_value(data)
+        if "portable_to" in data.keys():
+            reachable_from_playlist_ids = data.get("portable_to")
+            for playlist_id in reachable_from_playlist_ids:
+                try:
+                    Playlist.objects.get(pk=playlist_id)
+                except Playlist.DoesNotExist as error:
+                    raise serializers.ValidationError(
+                        {"playlist_not_exists": _("Playlist not exists.")}
+                    ) from error
+                if playlist_id == str(self.instance.id):
+                    raise serializers.ValidationError(
+                        {"cannot_port_to_self": _("Playlist i not portable to itself.")}
+                    )
+            value.update({"portable_to": reachable_from_playlist_ids})
+        return value
```

Comment:
I think you can improve greatly by not getting the whole object N times. And I would say the key of the validation error should be the name of the field in error:
```suggestion
            for playlist_id in reachable_from_playlist_ids:
                if not Playlist.objects.filter(pk=playlist_id).exists():
                    raise serializers.ValidationError(
                        {"portable_to": _("Playlist not exists.")}
                    ) from error
                if playlist_id == str(self.instance.id):
                    raise serializers.ValidationError(
                        {"portable_to": _("Playlist is not portable to itself.")}
                    )
            value.update({"portable_to": reachable_from_playlist_ids})
        return value
```

But I think it's doable in one query:
```suggestion
           if str(self.instance.id) in reachable_from_playlist_ids:
                raise serializers.ValidationError(
                    {"portable_to": _("Playlist is not portable to itself.")}
                )
            existing_playlist_ids = Playlist.objects.filter(pk__in=reachable_from_playlist_ids).values_list("id", flat=True)
            ids_diff = set(reachable_from_playlist_ids) - set(existing_playlist_ids)
            if ids_diff:
                raise serializers.ValidationError(
                    {"portable_to": _(f"Playlist not exists: {ids_diff!s}"))}
                ) from error
            value.update({"portable_to": reachable_from_playlist_ids})
        return value
```